includes/collection/class-following.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Following collection file.
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub\Collection;
+
+use Activitypub\Http;
+
+/**
+ * ActivityPub Following Collection.
+ */
+class Following {
+	/**
+	 * Meta key for the followers user ID.
+	 *
+	 * @var string
+	 */
+	const FOLLOWING_META_KEY = '_activitypub_followed_by';
+
+	/**
+	 * Meta key for pending followers user ID.
+	 *
+	 * @var string
+	 */
+	const PENDING_META_KEY = '_activitypub_followed_by_pending';
+
+	/**
+	 * Follow a user.
+	 *
+	 * @param \WP_Post|int $post    The ID of the remote Actor.
+	 * @param int          $user_id The ID of the WordPress User.
+	 *
+	 * @return \WP_Post|\WP_Error The ID of the Actor or a WP_Error.
+	 */
+	public static function follow( $post, $user_id ) {
+		$post = \get_post( $post );
+
+		if ( ! $post ) {
+			return new \WP_Error( 'activitypub_remote_actor_not_found', 'Remote actor not found' );
+		}
+
+		$following = \get_post_meta( $post->ID, self::FOLLOWING_META_KEY, false );
+
+		if ( ! \is_array( $following ) ) {
+			$following = array();
+		}
+
+		$pending = \get_post_meta( $post->ID, self::PENDING_META_KEY, false );
```

Comment:
```suggestion
		$all_meta  = get_post_meta( $post->ID );
		$following = $all_meta[ self::FOLLOWING_META_KEY ] ?? array();
		$pending   = $all_meta[ self::PENDING_META_KEY ] ?? array();
```
I don't know if this is helpful, performance-wise it probably doesn't make a difference, with meta caching and all, but might help with adding a fallback for empty pending key?

Reply:
let's try it!